lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_download_translations_action.rb

```diff
@@ -0,0 +1,92 @@
+# This action is the new version of android_update_metadata (AndroidUpdateMetadataAction) and should now be used instead of that one
+
+module Fastlane
+  module Actions
+    class AndroidDownloadTranslationsAction < Action
+      def self.run(params)
+        require_relative '../../helper/android/android_localize_helper.rb'
+        require_relative '../../helper/git_helper.rb'
+
+        res_dir = File.join(ENV['PROJECT_ROOT_FOLDER'] || '.', params[:project_dir_name], 'src', 'main', 'res')
+
+        Fastlane::Helper::Android::LocalizeHelper.create_available_languages_file(
+          res_dir: res_dir,
+          locale_codes: [params[:source_locale]] + params[:locales].map { |h| h[:android] }
+        )
+        Fastlane::Helper::Android::LocalizeHelper.download_from_glotpress(
+          res_dir: res_dir,
+          glotpress_project_url: params[:glotpress_url],
+          locales_map: params[:locales]
+        )
+
+        # Update submodules then lint translations
+        Fastlane::Helper::GitHelper.update_submodules()
+        Action.sh('./gradlew', params[:lint_task])
+
+        Fastlane::Helper::GitHelper.commit(message: 'Update translations', files: res_dir, push: true)
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Download Android string.xml files from GlotPress and lint the updated translations'
+      end
+
+      def self.details
+        'Download translations from GlotPress, update local strings.xml files accordingly, lint, and commit the changes'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :project_dir_name,
+            env_name: 'PROJECT_NAME',
+            description: 'The name of the Android project (i.e. the name of the parent folder containing `src/main/res`)',
+            type: String
+          ),
```

Comment:
Side thought: what would you think if I replaced this parameter/ConfigItem to be `res_dir` instead? I feel that it would make more sense when taking that action in isolation (as it focuses/works only on that dir in practice), wdyt @mokagio ? 🤔 

```suggestion
          FastlaneCore::ConfigItem.new(
            key: :res_dir,
            env_name: 'FL_DOWNLOAD_TRANSLATIONS_RES_DIR',
            description: "The path to the Android project's `res` dir (i.e. the one containing the `values*` subdirs)",
            type: String,
            default_value: "#{ENV['PROJECT_NAME']/src/main/res"
          ),
```

Reply:
I think this is a great idea.

I really like isolated actions. It would really be great if other people could use these so the more generic we make them (without compromising our internal processes) the better. Obviously, for someone to use this lane they would also need to use GlotPress to manage the localization of their app, which might not be very common. Still, the idea has lots of merit 👍

Reply:
Fixed in 257bcc1